pkg/loki/modules.go

```diff
@@ -197,6 +202,16 @@ func (t *Loki) initIngester() (err error) {
 	t.cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.Multi.ConfigProvider = multiClientRuntimeConfigChannel(t.runtimeConfig)
 	t.cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.MemberlistKV = t.memberlistKV.GetMemberlistKV
 	t.cfg.Ingester.LifecyclerConfig.ListenPort = &t.cfg.Server.GRPCListenPort
+
+	// We want ingester to also query the store when using boltdb
+	if ActiveIndexType(t.cfg.SchemaConfig) == "boltdb" {
+		t.cfg.Ingester.QueryStore = true
+		// When using archiver, limit max look back for query to MaxChunkAge + upload interval by archiver + 15 mins to query only data whose index is not pushed yet
```

Comment:
Can you explain the why we need a look back parameter ?

Reply:
In microservices mode it would query for data which ingester itself has flushed while in single binary mode it would query all the data for whole query duration. This would mean there would be duplicates. I just set lookback period just enough to not miss any data.
What do you think?

Reply:
I think there are 2 use cases here around a filesystem and we should figure out what we want to support:

* A shared filesystem (like SAN or NFS)
* A non-shared filesystem (each instance running on it's own machine for example)

In the second use case each ingester would always be responsible for querying all of the data, but in the first case this would be treated just like s3 or gcs... i'm almost wondering if we should just define them as different store types to be able to use in the logic here for setting max QueryStoreMaxLookBack?

Reply:
The other concern is when we are using a shared object store, how much duplicate data are we going to get from each ingester, and is there a way to limit them to only using the index files they generate when querying the store?

Reply:
So there are 2 things here wrt storage, chunks and index files. I would say it is safe to assume chunks would always be in a shared filesystem otherwise queriers would not be able to fetch them. For index, it should not matter to differentiate because we anyways keep syncing the files and we just need to query ingesters for live data that they have not uploaded or just uploaded, which we already are doing here. I might be wrong here but does it make sense?

Reply:
I'd like to support the case where chunks/index are not on a shared filesystem, I think we can do this easy enough if the ingester doesn't force this limited lookback but instead can do a lookback of all the data, WDYT?

Reply:
It is doable but I think it would be better to have a config instead of inferring whether the storage is shared or not. We can expose store look back config in ingesters which would default to getting set to what is now.